src/FontManager.ts

```diff
@@ -74,35 +88,21 @@ export class FontManager extends EventDispatcher<EventMap> {
 
   update(newChars: string): void {
     let needsUpdate = false;
+    let newAlphabet = "";
     for (const char of newChars) {
       if (!this.alphabet.includes(char)) {
-        this.alphabet += char;
+        newAlphabet += char;
         needsUpdate = true;
       }
     }
 
     if (!needsUpdate) {
       return;
     }
-    const atlasWidth = 1024;
-    const atlasHeight = 1024;
-    const atlas = new Uint8ClampedArray(atlasWidth * atlasHeight);
-    const fontSize = this.options.fontSize ?? 48;
-    const buffer = Math.ceil(fontSize / 16);
-    const tinysdf = new TinySDF({
-      fontSize,
-      buffer,
-      radius: Math.ceil(fontSize / 4),
-      fontFamily: this.options.fontFamily ?? "monospace",
-    });
+    const atlas = this.atlasData.data;
 
-    const charInfo: Record<string, CharInfo> = {};
-    let x = 0;
-    let y = 0;
-    let rowHeight = 0;
-    let lineHeight = 0;
-    let maxAscent = 0;
-    for (const char of this.alphabet) {
+    const charInfo: Record<string, CharInfo> = this.atlasData.charInfo;
+    for (const char of newAlphabet) {
       if (charInfo[char] != undefined) {
```

Comment:
🤔 is this even possible? one would think the `includes` checks would avoid this scenario, right?